### PR TITLE
Add struct and lock around health check info map.

### LINKED ISSUE #2920: Data race in health checks
**Branch/Environment/Version**
- Branch/Version: `	github.com/TykTechnologies/tyk v1.9.2-0.20200309170108-22e6f2f117fe` module used by `tyk-analytics`

**Describe the bug**
Test case `TestAPIContext_UpdateKey/Update_basic_token_with_username` in `tyk-analytics` would occasionally fail with a data race.
From my very brief analysis, about three out of every hundred executions.

**Reproduction steps**
Run `TestAPIContext_UpdateKey/Update_basic_token_with_username` in `tyk-analytics` until it falls over..?

**Actual behavior**
Occasional failure within `tyk/gateway/health_check.go` due to concurrent map access.

**Expected behavior**
Concurrent map access should be protected with a mutex lock.

**Screenshots/Video**
N/A

**Logs (debug mode or log file):**
[Most recent CI failure log](https://app.buddy.works/tyk-projects/tyk-analytics/pipelines/pipeline/237268/execution/5e689474422f5a1a30810947/action/523410/logs).

**Configuration (tyk config file):**
N/A - endemic to test case

**Additional context**
This issue was exposed while investigating https://github.com/TykTechnologies/tyk-analytics/pull/1780.

### PR DESCRIPTION
**Pull request by jlucktay**
Fixes #2920.

- [x] Fork the repo and create your branch from the **latest** `master`.
- [x] If you've added code that should be tested, add tests!
  - failing test driving this change is in https://github.com/TykTechnologies/tyk-analytics
- [x] If you've changed APIs, describe what needs to be updated in the documentation.
  - no API changes
- [x] Ensure the test suite passes (`go test`).
  - tests are passing ✅
- [x] Make sure your code lints (`go fmt && go vet`).
  - `fmt` and `vet` both OK ✅

### CODE REVIEW

**Comment by buger** [comment]
I wonder if it just makes sense using sync.Map here instead.

**Comment by jlucktay** [comment]
I [did some reading](https://medium.com/@deckarep/the-new-kid-in-town-gos-sync-map-de24a6bf7c2c) about the new `sync.Map` a little while ago.
We can give it a go. 🤷‍♂

**Comment by jlucktay** [comment]
Tangent: [this](https://www.youtube.com/watch?v=C1EtfDnsdDs) is a pretty good summary of `sync.Map` by one of its Go team implementors.

Those CI failures seem like they're in areas unrelated to this, and are a bit flaky/unreliable. Quite a lot of red on the Travis history page lately. 🤔

**Comment by jlucktay** [comment]
Just saw the conflict note and it seems like @sredxny has implemented the same thing I did the first time around... 😅

File: gateway/health_check.go [hunk 1]
```diff
@@ -35,16 +35,33 @@ var (
...
 
 func getHealthCheckInfo() map[string]HealthCheckItem {
 	healthCheckLock.Lock()
-	ret := healthCheckInfo.Load().(map[string]HealthCheckItem)
+	syncMap := healthCheckInfo.Load()
 	healthCheckLock.Unlock()
+
+	sm, ok := syncMap.(sync.Map)
+	if !ok {
+		panic("stored health check info is of incorrect type")
```
Comment (alephnull) [inline gateway/health_check.go]: Not sure if a panic is appropriate here. I might be missing some context on how the healthcheck operates though.

Comment (jlucktay) [reply gateway/health_check.go]: Thanks for taking a look mate; 1d4890c9 resolves by stepping it down to an error log call with the relevant health check info included

**Comment by buger** [comment]
@jlucktay am I right that this PR can be closed, since it was already fixed by @sredxny changes?

**Comment by jlucktay** [comment]
> @jlucktay am I right that this PR can be closed, since it was already fixed by @sredxny changes?

Yeah, we can bin this one off now, it's basically a duplicate.

**Pull request closed without merging**